File: hotel_booking/v2/post/Java SDK/HotelBooking.java
import com.amadeus.Amadeus;
import com.amadeus.exceptions.ResponseException;
import com.amadeus.resources.HotelOrder;

public class FlightSearch {

  public static void main(String[] args) throws ResponseException {

    Amadeus amadeus = Amadeus
        .builder("YOUR_AMADEUS_CLIENT_ID","YOUR_AMADEUS_CLIENT_SECRET")
        .build();

    String body = "{\n" +
        "  \"data\": {\n" +
        "    \"type\": \"hotel-order\",\n" +
        "    \"guests\": [\n" +
        "      {\n" +
        "        \"tid\": 1,\n" +
        "        \"title\": \"MR\",\n" +
        "        \"firstName\": \"BOB\",\n" +
        "        \"lastName\": \"SMITH\",\n" +
        "        \"phone\": \"+33679278416\",\n" +
        "        \"email\": \"bob.smith@email.com\"\n" +
        "      }\n" +
        "    ],\n" +
        "    \"travelAgent\": {\n" +
        "      \"contact\": {\n" +
        "        \"email\": \"bob.smith@email.com\"\n" +
        "      }\n" +
        "    },\n" +
        "    \"roomAssociations\": [\n" +
        "      {\n" +
        "        \"guestReferences\": [\n" +
        "          {\n" +
        "            \"guestReference\": \"1\"\n" +
        "          }\n" +
        "        ],\n" +
        "        \"hotelOfferId\": \"HL2E6A0IQZ\"\n" +
        "      }\n" +
        "    ],\n" +
        "    \"payment\": {\n" +
        "      \"method\": \"CREDIT_CARD\",\n" +
        "      \"paymentCard\": {\n" +
        "        \"paymentCardInfo\": {\n" +
        "          \"vendorCode\": \"VI\",\n" +
        "          \"cardNumber\": \"4151289722471370\",\n" +
        "          \"expiryDate\": \"2026-08\",\n" +
        "          \"holderName\": \"BOB SMITH\"\n" +
        "        }\n" +
        "      }\n" +
        "    }\n" +
        "  }\n" +
        "}";
    
    HotelOrder hotel = amadeus.booking.hotelOrders.post(body);

    if(hotel.getResponse().getStatusCode() != 201) {
      System.out.println("Wrong status code: " + hotel.getResponse().getStatusCode());
      System.exit(-1); 
    }
    
    System.out.println(hotel);
  }

}

File: hotel_booking/v1/post/Android SDK/hotel_booking.kt
// How to install the library at https://github.com/amadeus4dev/amadeus-android

import com.amadeus.android.Amadeus
import com.amadeus.android.ApiResult
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.launch
import android.util.Log;

val job = SupervisorJob()
val scope = CoroutineScope(Dispatchers.Main + job)

val amadeus = Amadeus.Builder(context)
    .setClientId("REPLACE_BY_YOUR_API_KEY")
    .setClientSecret("REPLACE_BY_YOUR_API_SECRET")
    .build()

val body = "{" +
                            "  \"data\": {" +
                            "    \"offerId\": \"8MXZ1TKO5T\"," +
                            "    \"guests\": [" +
                            "      {" +
                            "        \"id\": 1," +
                            "        \"name\": {" +
                            "          \"title\": \"MR\"," +
                            "          \"firstName\": \"BOB\"," +
                            "          \"lastName\": \"SMITH\"" +
                            "        }," +
                            "        \"contact\": {" +
                            "          \"phone\": \"+33679278416\"," +
                            "          \"email\": \"bob.smith@email.com\"" +
                            "        }" +
                            "      }" +
                            "    ]," +
                            "    \"payments\": [" +
                            "      {" +
                            "        \"id\": 1," +
                            "        \"method\": \"creditCard\"," +
                            "        \"card\": {" +
                            "          \"vendorCode\": \"VI\"," +
                            "          \"cardNumber\": \"4151289722471370\"," +
                            "          \"expiryDate\": \"2021-08\"" +
                            "        }" +
                            "      }" +
                            "    ]," +
                            "    \"rooms\": [" +
                            "      {" +
                            "        \"guestIds\": [" +
                            "          1" +
                            "        ]," +
                            "        \"paymentId\": 1," +
                            "        \"specialRequest\": \"I will arrive at midnight\"" +
                            "      }\n" +
                            "    ]" +
                            "  }" +
                            "}"    

scope.launch {
  when (val booking = amadeus.booking.hotelBooking.post(body)) {
    is ApiResult.Success -> {
      Log.d("Result", "${result.data}")
    }
    is ApiResult.Error -> {
      // Handle your error
    }
  }
}
